adding partial file (#3624)
fishtown-analytics/docs.getdbt.com: contributing/single-sourcing-content.md

<Var name="dbt" /> is awesome!
```

## Reusing content

Use some techniques to share the same content on different pages, like partial files or snippets. 

Using partial files, which is a built-in Docusaurus feature, is recommended over snippets.

### Partial

A partial file allows you to reuse content throughout the Docs. Here are the steps you can take to create and use a partial file:

1. Create a new markdown partial file in the `website/snippets` directory. The file name must begin with an underscore, like `_filename.md`
2. Go back to the Docs file that's pulling content from the partial file 
3. Add the following import file: `import SetUpPages from '/snippets/_this-is-your-partial-file-name.md';`
   * To use partials, you must always add an import file in that format. Note `SetUpPages` can be whatever makes sense for your purpose.
   * `.md` needs to be added to the end of the filename. 

> 3. Go to the next line and add `<SetUpPages anyname={'/snippets/_this-is-your-partial-file-name.md'}/>`. This fetches the reusable content in the partial file

Reviewer: For this, step 2 looks correct with the import statement. 

For step 3, when using the partial component, you don't have to include the file path within a prop. The component can be used as `<SetUpPages />` without any props.

If you were passing in data into the partial, that is when we would use a prop on the component. This is useful when using dynamic data (for example if you wanted to use the same partial on multiple docs pages, but changing the values within the partial for each page).

Looking at the example in the [Docusaurus docs](https://docusaurus.io/docs/markdown-features/react#importing-markdown), if we wanted to pass in a `name` into the partial, we would do:

```
import PartialExample from '/snippets/_markdown-partial-example.mdx';

<PartialExample name="Sebastien" />
```

And within the partial file itself, the name prop would be available at `props.name`:
```
<span>Hello {props.name}</span>

This is demo content.
```

   * Note `anyname` can be whatever makes sense for your purpose.

You can also use this for more advanced use cases like reusable frontmatter.

#### Partial example

To create a new partial to use throughout the site, first, we will create a new markdown partial file within the snippets directory:

```markdown
/snippets/_partial-name.md
## Header 2

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam fermentum porttitor dui, id scelerisque enim scelerisque at.
```

Now, we can add this snippet to a Docs file with the Snippet component:

```markdown
Docs content here.

> import SetUpPages from 'import SetUpPages from '/snippets/_this-is-your-partial-file-name.md';

Author: ```suggestion
`import SetUpPages from '/snippets/_this-is-your-partial-file-name.md';`
```

> <SetUpPages anyname={'/snippets/_this-is-your-partial-file-name.md'}/>

Reviewer: Same note as above, this can be structured as:

```markdown
Docs content here.

import SetUpPages from '/snippets/_this-is-your-partial-file-name.md';

<SetUpPages />

Docs content here.
```

2. Go back to the docs file that will pull content from the partial file.
3. Add the following import file: `import ComponentName from '/snippets/_this-is-your-partial-file-name.md';`
   * You must always add an import file in that format. Note you can name `ComponentName` (a partial component) can be whatever makes sense for your purpose.
   * `.md` needs to be added to the end of the filename. 
4. To use the partial component, go to the next line and add `<ComponentName />`. This fetches the reusable content in the partial file
   * Note `anyname` can be whatever makes sense for your purpose.

You can also use this for more advanced use cases like reusable frontmatter.

#### Partial example

To create a new partial to use throughout the site, first, we will create a new markdown partial file within the snippets directory:

```markdown
/snippets/_partial-name.md
## Header 2

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam fermentum porttitor dui, id scelerisque enim scelerisque at.
```

Now, we can add this code to a Docs file:

```markdown
Docs content here.

`import SetUpPages from '/snippets/_partial-name.md';`

<SetUpPages />

Docs content here.
```

> <details>

Author: hey @JKarlavige , ok i've reworded this based on your feedback and also added a toggle with info on how to use props to pass diff content. whenever you have a chance in the upcoming week, can you give this a 👀 to see if it's correct? thank you!

   * `.md` needs to be added to the end of the filename. 
4. To use the partial component, go to the next line and add `<ComponentName />`. This fetches the reusable content in the partial file
   * Note `anyname` can be whatever makes sense for your purpose.

You can also use this for more advanced use cases like reusable frontmatter.

#### Partial example

1. To create a new partial to use throughout the site, first, we will create a new markdown partial file within the snippets directory:

```markdown
/snippets/_partial-name.md
```

2. Add the following reusable content in the `/snippets/_partial-name.md` partial file:

```markdown
## Header 2

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam fermentum porttitor dui, id scelerisque enim scelerisque at.
```

3. Now, go back to the docs file and add the following code to fetch the reusable content added in the partial file:

```markdown
Docs content here.

`import SetUpPages from '/snippets/_partial-name.md';`

<SetUpPages />        					

Docs content here.
```

- `import SetUpPages from '/snippets/_partial-name.md';`  &mdash; A partial file that will be imported by other files

> - `<SetUpPages />` &mdash;  A component that passes in data into the partial. In this example, the component is passing 'feature' into the partial. This is useful when using dynamic data (for example if you wanted to use the same partial on multiple docs pages, but change the values within the partial for each page)

Author: ```suggestion
- `<SetUpPages />` &mdash;  A component that imports content from the partial file. You can also use it to pass in data into the partial using props (See 'How to use props to pass different content on multiple pages?' below). 
```

   * Note `anyname` can be whatever makes sense for your purpose.

You can also use this for more advanced use cases like reusable frontmatter.

#### Partial example

1. To create a new partial to use throughout the site, first, we will create a new markdown partial file within the snippets directory:

```markdown
/snippets/_partial-name.md
```

2. Add the following reusable content in the `/snippets/_partial-name.md` partial file:

```markdown
## Header 2

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam fermentum porttitor dui, id scelerisque enim scelerisque at.
```

3. Now, go back to the docs file and add the following code to fetch the reusable content added in the partial file:

```markdown
Docs content here.

`import SetUpPages from '/snippets/_partial-name.md';`

<SetUpPages />        					

Docs content here.
```

- `import SetUpPages from '/snippets/_partial-name.md';`  &mdash; A partial file that will be imported by other files
- `<SetUpPages />` &mdash;  A component that imports content from the partial file. You can also use it to pass in data into the partial using props (See 'How to use props to pass different content on multiple pages?' below). 

> 3. This will then render the content of the docs in the partial file.

Author: ```suggestion
4. This will then render the content of the docs in the partial file.
```
